### PR TITLE
Grid examples

### PR DESCRIPTION
**Pull request by rachelandrew**
A good chunk of the grid properties, also the `repeat` function which only makes sense in a grid layout context but I wanted to check that one was correct.

### CODE REVIEW

File: live-examples/css-examples/grid/grid-auto-flow.html [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    <div class="example-choice">
+        <pre><code class="language-css">grid-auto-flow: column;</code></pre>
+        <button type="button" class="copy hidden" aria-hidden="true">
+            <span class="visually-hidden">Copy to Clipboard</span>
+        </button>
+    </div>
+
+    <div class="example-choice">
+        <pre><code class="language-css">grid-auto-flow: sparse;</code></pre>
```
Comment (wbamberg) [inline live-examples/css-examples/grid/grid-auto-flow.html]: I can't find `sparse` as a value in the MDN docs page or the spec, and it doesn't seem to be supported in Firefox or Chrome.

File: live-examples/css-examples/grid/meta.json [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        "gridTemplateColumns": {
+            "baseTmpl": "tmpl/live-css-tmpl.html",
+            "cssExampleSrc":
+                "../../live-examples/css-examples/grid/grid-template-columns.css",
+            "exampleCode":
+                "live-examples/css-examples/grid/grid-template-columns.html",
+            "fileName": "grid-template-columns.html",
+            "title": "CSS Demo: grid-template-columns",
+            "type": "css"
+        },
```
Comment (wbamberg) [inline live-examples/css-examples/grid/meta.json]: It's helpful to have the entries listed alphabetically.

File: live-examples/css-examples/grid/meta.json [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        "repeat": {
+            "baseTmpl": "tmpl/live-css-tmpl.html",
+            "cssExampleSrc":
+                "../../live-examples/css-examples/grid/function-repeat.css",
+            "exampleCode":
+                "live-examples/css-examples/grid/function-repeat.html",
+            "fileName": "function-repeat.html",
+            "title": "CSS Demo: repeat",
```
Comment (wbamberg) [inline live-examples/css-examples/grid/meta.json]: We should include brackets for functions: `CSS Demo: repeat()`, to match the title.

File: live-examples/css-examples/grid/grid-auto-flow.html (hunk 1 above)
Comment (rachelandrew) [reply live-examples/css-examples/grid/grid-auto-flow.html]: you are indeed correct, sparse is the default packing mode, it doesn't actually have a keyword. Some sort of brain autocomplete issue :)

Comment (rachelandrew) [reply live-examples/css-examples/grid/grid-auto-flow.html]: and good call on making the changed element a different colour, that really helps.

File: live-examples/css-examples/grid/meta.json (hunk 2 above)
Comment (rachelandrew) [reply live-examples/css-examples/grid/meta.json]: done

File: live-examples/css-examples/grid/grid-column-start.html [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    <div class="example-choice">
+        <pre><code class="language-css">grid-column-start: 2;</code></pre>
+        <button type="button" class="copy hidden" aria-hidden="true">
+            <span class="visually-hidden">Copy to Clipboard</span>
+        </button>
+    </div>
+
+    <div class="example-choice">
+        <pre><code class="language-css">grid-column-start: -1;</code></pre>
```
Comment (schalkneethling) [inline live-examples/css-examples/grid/grid-column-start.html]: Would it be useful here to also have an example of `grid-column-start: span 3;` as you did for `grid-column-end`?

File: live-examples/css-examples/grid/grid-row-start.html [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+    <div class="example-choice">
+        <pre><code class="language-css">grid-row-start: -1;</code></pre>
+        <button type="button" class="copy hidden" aria-hidden="true">
+            <span class="visually-hidden">Copy to Clipboard</span>
+        </button>
+    </div> 
+
+    <div class="example-choice">
+        <pre><code class="language-css">grid-row-start: 3;</code></pre>
```
Comment (schalkneethling) [inline live-examples/css-examples/grid/grid-row-start.html]: This is a duplicate of line 10 above

File: live-examples/css-examples/grid/grid-row-start.css [hunk 6]
```diff
@@ -0,0 +1,20 @@
+.example-container {
+    background-color: #eee;
+    border: .75em solid;
+    padding: .75em;
+    display: grid;
+    grid-template-columns: 1fr 1.5fr 1fr;
+    grid-template-rows: repeat(3,minmax(40px, auto));
```
Comment (schalkneethling) [inline live-examples/css-examples/grid/grid-row-start.css]: Nit: space after `3,`

File: live-examples/css-examples/grid/grid-row.css [hunk 7]
```diff
@@ -0,0 +1,20 @@
+.example-container {
+    background-color: #eee;
+    border: .75em solid;
+    padding: .75em;
+    display: grid;
+    grid-template-columns: 1fr 1.5fr 1fr;
+    grid-template-rows: repeat(3,minmax(40px, auto));
```
Comment (schalkneethling) [inline live-examples/css-examples/grid/grid-row.css]: Nit: space after `3,`

File: live-examples/css-examples/grid/grid-row-end.css [hunk 8]
```diff
@@ -0,0 +1,20 @@
+.example-container {
+    background-color: #eee;
+    border: .75em solid;
+    padding: .75em;
+    display: grid;
+    grid-template-columns: 1fr 1.5fr 1fr;
+    grid-template-rows: repeat(3,minmax(40px, auto));
```
Comment (schalkneethling) [inline live-examples/css-examples/grid/grid-row-end.css]: Nit: space after `3,`

File: live-examples/css-examples/grid/grid-column.css [hunk 9]
```diff
@@ -0,0 +1,20 @@
+.example-container {
+    background-color: #eee;
+    border: .75em solid;
+    padding: .75em;
+    display: grid;
+    grid-template-columns: 1fr 1.5fr 1fr;
+    grid-template-rows: repeat(3,minmax(40px, auto));
```
Comment (schalkneethling) [inline live-examples/css-examples/grid/grid-column.css]: Nit: space after `3,`

File: live-examples/css-examples/grid/grid-column-start.css [hunk 10]
```diff
@@ -0,0 +1,20 @@
+.example-container {
+    background-color: #eee;
+    border: .75em solid;
+    padding: .75em;
+    display: grid;
+    grid-template-columns: 1fr 1.5fr 1fr;
+    grid-template-rows: repeat(3,minmax(40px, auto));
```
Comment (schalkneethling) [inline live-examples/css-examples/grid/grid-column-start.css]: Nit: space after `3,`

File: live-examples/css-examples/grid/grid-column-end.css [hunk 11]
```diff
@@ -0,0 +1,20 @@
+.example-container {
+    background-color: #eee;
+    border: .75em solid;
+    padding: .75em;
+    display: grid;
+    grid-template-columns: 1fr 1.5fr 1fr;
+    grid-template-rows: repeat(3,minmax(40px, auto));
```
Comment (schalkneethling) [inline live-examples/css-examples/grid/grid-column-end.css]: Note: space after `3,`

File: live-examples/css-examples/grid/grid-auto-flow.css [hunk 12]
```diff
@@ -0,0 +1,23 @@
+#example-element {
+    background-color: #eee;
+    border: .75em solid;
+    padding: .75em;
+    display: grid;
+    grid-template-columns: 1fr 1fr 1fr;
+    grid-template-rows: repeat(3,minmax(40px, auto));
```
Comment (schalkneethling) [inline live-examples/css-examples/grid/grid-auto-flow.css]: Nit: space after `3,`

**Pull request merged**